### ISSUE TITLE
failing xgboost tests

### ISSUE THREAD

**Issue by derenrich** [TeamHG-Memex/eli5#259]
So I'm seeing xgboost tests as failing. I started investigating this because I couldn't get the titanic notebook example to run as provided. Then I couldn't get the test suite to run through tox as provided. I had to apply the following patch.

```
diff --git a/docs/source/conf.py b/docs/source/conf.py
index 0d955af..5b7ba71 100644
--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -35,22 +35,6 @@ class Mock(MagicMock):
         return Mock()


-MOCK_MODULES = [
-    'lightning',
-    'lightning.impl',
-    'lightning.impl.base',
-    'sklearn_crfsuite',
-    # 'IPython',
-    # 'IPython.display',
-    # 'IPython.lib',
-    # 'typing',
-    'xgboost',
-    'lightgbm',
-    'pandas',
-    'pytest',
-]
-sys.modules.update((mod_name, Mock()) for mod_name in MOCK_MODULES)
-

 # -- General configuration ------------------------------------------------
diff --git a/tox.ini b/tox.ini
index 6b82311..ebc9039 100644
--- a/tox.ini
+++ b/tox.ini
@@ -17,7 +17,8 @@ deps=
     numpy
     scipy
     scikit-learn
-
+    mock
+    sphinx-rtd-theme

 [testenv]
 whitelist_externals = /bin/bash
```

But once I did that the xgboost tests started failing en masse. I'd guess the underlying API changed under you but looking at xgboost's source history that doesn't seem to be the case so I'm pretty confused.

**Comment by derenrich** [comment]
it's the same exception over and over again.

```
xgb = XGBRegressor(base_score=0.5, booster='gbtree', colsample_bylevel=1,
       col..., reg_lambda=1, scale_pos_weight=1, seed=None,
       silent=True, subsample=1), is_regression = None

    def _check_booster_args(xgb, is_regression=None):
        # type: (Any, bool) -> Tuple[Booster, bool]
        if isinstance(xgb, Booster):
            booster = xgb
        else:
>           booster = xgb.booster()
E           TypeError: 'str' object is not callable

eli5/xgboost.py:225: TypeError
```

**Comment by haribaskar** [comment]
# TypeError: 'str' object is not callable
### prediction_text = explain_prediction(model,X_test.iloc[i])

File "/anaconda/envs/py35/lib/python3.5/site-packages/singledispatch.py", line 210, in wrapper

    return dispatch(args[0].__class__)(*args, **kw)

  File "/home/azure-user/.local/lib/python3.5/site-packages/eli5/xgboost.py", line 148, in explain_prediction_xgboost

    booster, is_regression = _check_booster_args(xgb, is_regression)

  File "/home/azure-user/.local/lib/python3.5/site-packages/eli5/xgboost.py", line 225, in _check_booster_args

    booster = xgb.booster()

TypeError: 'str' object is not callable

## After we changing xgb.get_booster() in xgboost.py the issue got solved.

**Comment by haribaskar** [comment]
same issue #252

**Comment by lopuhin** [comment]
#268 should fix this once it's merged

**Comment by lopuhin** [comment]
fixed in #268
